backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/dialect/impl/ElasticsearchDialectFactory.java

```diff
@@ -78,6 +78,7 @@ public ElasticsearchModelDialect createModelDialect(ElasticsearchVersion version
 
 	private ElasticsearchModelDialect createModelDialectElastic(ElasticsearchVersion version) {
 		OptionalInt majorOptional = version.majorOptional();
+		OptionalInt minorOptional = version.minor();
```

Comment:
I think you forgot about protocol dialects?

Also, aren't there some methods in `TckBackendFeatures` that you could remove because they now always return the same value for all supported versions?